### PR TITLE
initialize feature lifecycle doc

### PR DESCRIPTION
**Pull request by kevin-wangzefeng**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design

/kind documentation
> /kind test
> /kind failing-test
> /kind feature

**What this PR does / why we need it**:
clarify definitions and differences between features and corresponding APIs during different development stages (versions).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/getting-started/feature-lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Feature L`ifecycle
```
Comment (rohitsardesai83) [inline docs/getting-started/feature-lifecycle.md]: L`ifecycle -> Lifecycle

File: docs/getting-started/feature-lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+This document is to clarify definitions and differences between features and corresponding APIs
+during different development stages (versions).
+
+Each version haves different level of stability, support time,
```
Comment (rohitsardesai83) [inline docs/getting-started/feature-lifecycle.md]: Each version ~haves~ has different level of

File: docs/getting-started/feature-lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+* [GA](#ga)
+
+
+## Alpha
+
+The feature may be changed/upgraded in incompatible ways in the later versions.
+
+The source code will be available in the release branch/tag as well as in the binaries.
+
+Support for the feature can be stopped any time without notice
```
Comment (rohitsardesai83) [inline docs/getting-started/feature-lifecycle.md]: end with a period.

File: docs/getting-started/feature-lifecycle.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+The feature will not induce bugs in other APIs/Features if enabled.
+
+The feature will be completely implemented.
+
+The API version names will be like v1beta1, v1beta2, etc. The suffixed number will be incremented by 1 in each upgrade.
+
+### Graduation Criteria
+-  Should have at least 50% coverage in e2e tests.
+-  Project agrees to support this feature for at least next 2 minor releases and notice of at least 2 minor releases will be given before stopping the support.
+-  Feature Owner should commit to ensure backward/forward. compatibility in the later versions.
```
Comment (rohitsardesai83) [inline docs/getting-started/feature-lifecycle.md]: remove period after the word "forward"

**Comment by subpathdev** [comment]
The life cycle looks good to me;

**Comment by kevin-wangzefeng** [comment]
Fixed, PTAL, thanks.

**Pull request merged**